elastic/kibana
# [ResponseOps] Disabling flapping setting makes the recovered alerts to stay in active state

Resolves https://github.com/elastic/kibana/issues/190609

## Summary

When we added the flapping functionality we started storing the recovered alerts in the task state. To save space we added an optimization in the function `getAlertsToSerialize` to not store recovered alerts in the task state when flapping in disabled as we don't need to track them. The alerts client is calling this function to get the active and recovered alerts we need to write to the alert documents. When flapping is disabled the function is not returning the recovered alerts and so the alert docs never get updated, and the alerts appear to stay active.

This PR updates that function to turn off the optimization when it's called from the alerts client.


### Checklist



### To verify

- Go to the rule settings and disable flapping
- Create an ES query rule
- Create an active alert and then let it recover
- Verify that the AAD doc is updated to be recovered, go to [dev tools](http://localhost:5601/app/dev_tools#/console) and run the following
```
GET .internal.alerts-stack.alerts-default*/_search
```
- Verify that the ui also indicates the alert is recovered
